app.rb

```diff
@@ -0,0 +1,171 @@
+require './person'
+require './book'
+require './rental'
+require './student'
+require './teacher'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def main
+    loop do
+      homepage
+      selected_option = gets.chomp.to_i
+      case selected_option
+      when 1 then list_all_books
+      when 2 then list_all_people
+      when 3 then create_teacher_or_student
+      when 4 then create_a_book
+      when 5 then create_rental
+      when 6 then list_rentals_for_a_person
+      when 7
+        puts 'Thanks for using this app'
+        exit
+      else puts 'Error: Enter a valid option.'
+      end
+    end
+  end
+
+  # rubocop:enable Metrics/CyclomaticComplexity
+  private
+
+  def homepage
+    puts 'Welcome to school library App'
+    puts 'Please choose an option by entering in a number'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person(teacher or student, not a plain person)'
+    puts '4. Create a book'
+    puts '5. Create rental'
+    puts '6. List rentals for a given person id'
+    puts "7. Quit\n"
+  end
+
+  def list_all_books
+    puts 'List of all books'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}\n"
+      puts
+    end
+  end
+
+  def list_all_people
+    puts 'List of all people'
+    @people.each do |person|
+      puts "[#{person.class}] name: #{person.name}, ID: #{person.id}, age: #{person.age}\n"
+      puts
+    end
+  end
```

Comment:
- [ ] Please your app can not list all people so make sure you fix this error :point_down: to get this method of listing people to work perfectly. :pray: 
  
  ![image](https://user-images.githubusercontent.com/92303891/233312118-d30f64ed-0e68-4865-8ac3-48adba0dbc00.png)